server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala

```diff
@@ -0,0 +1,394 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aliyun.emr.rss.service.deploy.worker
+
+import java.io.IOException
+import java.util.{ArrayList => jArrayList, List => jList}
+import java.util.concurrent._
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
+import java.util.function.BiFunction
+
+import scala.collection.JavaConverters._
+
+import io.netty.util.{HashedWheelTimer, Timeout, TimerTask}
+import io.netty.util.internal.ConcurrentSet
+
+import com.aliyun.emr.rss.common.RssConf
+import com.aliyun.emr.rss.common.internal.Logging
+import com.aliyun.emr.rss.common.meta.{PartitionLocationInfo, WorkerInfo}
+import com.aliyun.emr.rss.common.metrics.MetricsSystem
+import com.aliyun.emr.rss.common.protocol.{PartitionLocation, PartitionSplitMode}
+import com.aliyun.emr.rss.common.protocol.PartitionLocation.StorageHint
+import com.aliyun.emr.rss.common.protocol.message.ControlMessages._
+import com.aliyun.emr.rss.common.protocol.message.StatusCode
+import com.aliyun.emr.rss.common.rpc._
+import com.aliyun.emr.rss.common.util.Utils
+
+private[deploy] class Controller(
```

Comment:
Maybe `WorkerController` or `WorkerRPCControl`?  I am not sure if we will do same thing to master

Reply:
It's inside server-worker package, so I think it's unnecessary to add the Worker prefix.

Reply:
Again, I've considered this..